test/sqlite_perfdb.cpp

```diff
@@ -708,8 +708,8 @@ class DBMultiThreadedTestWork
 };
 
 unsigned int DBMultiThreadedTestWork::threads_count    = 16;
-unsigned int DBMultiThreadedTestWork::common_part_size = 32;
-unsigned int DBMultiThreadedTestWork::unique_part_size = 32;
+unsigned int DBMultiThreadedTestWork::common_part_size = 16;
```

Comment:
Reducing the parallelism can potentially mask errors which only show up in under stress.

Reply:
There is clearly something going on. Please help investigate.

Reply:
> There is clearly something going on. Please help investigate.

This was done to reduce work, not because there is an error. Please see description of the PR

Reply:
We have problems with perfdb tests (both file-based and sqlite-based). I was working on it when comgr arrived so I had to postpone it. Reducing load helps to reduce propability of test failure.

Reply:
How does the perf db error manifest ?

Reply:
Since these tests use a lot of threads we should set the `RUN_SERIAL` property in cmake for the tests instead of lowering the number of threads.

Reply:
```
terminate called after throwing an instance of 'miopen::Exception'
  what():  /var/jenkins/workspace/en_wrw-igemm-v4r4xdlops-fp32-fix/src/sqlite_db.cpp:114: Internal error while accessing SQLite database: UNIQUE constraint failed: perf_db.solver, perf_db.config, perf_db.arch, perf_db.num_cu
```

See http://micimaster.amd.com/blue/organizations/jenkins/MLLibs%2FMIOpen/detail/wrw-igemm-v4r4xdlops-fp32-fix/14/pipeline/13/ for an example.

Reply:
#227 fixes this.

Reply:
Great! Just revert https://github.com/ROCmSoftwarePlatform/MIOpen/commit/9154e70e2a7506d4454cf87678bcd2b2c28deb95 in your branch after rebasing to develop, when/if this PR will be merged in, and everything will be restored.

Reply:
I agree, lets get things in shape first, that PR can wait for now.

Reply:
@pfultz2 
> Since these tests use a lot of threads we should set the `RUN_SERIAL` property in cmake for the tests instead of lowering the number of threads.

Good to know, thanks, we'll make this change. The problem is that the test itself is unstable -- e.g. may fail even if run alone, depending on the compiler (clang or GCC), build options (`-O`) etc. Moreover I believe it shouldn't fail even when system is under high load, but this is not so(((